### PR TITLE
Fix for the #786

### PR DESCRIPTION
**Pull request by haouarihk**
FIxed the issue with decode #786

### CODE REVIEW

**Comment by haouarihk** [comment]
i hate macos 16 for this

**Comment by haouarihk** [comment]
i just corrected a little mistake @hikarunoryoma made with 
`Buffer.from` he did `buffer.from(base64).toString("base64")` which essentially takes in the base64 as a utf-8 and tries to convert it into a string of type "base64"

but instead it should be
`buffer.from(base64, "base64).toString()` which is telling the buffer object to take that base64 and tell it it's a type of base64, and then stringify that buffer

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,9 @@
 coverage/
 node_modules/
 test/output/
+
+# ignore nyc output
+.nyc_output
+
+# ignore history
+.history
```
Comment (marco-c) [inline .gitignore:9]: These changes in .gitignore are OK, but can you create a separate PR for them? I prefer to have self-contained PRs.

File: src/vapid-helper.js [hunk 2]
```diff
@@ -100,14 +100,14 @@ function validatePublicKey(publicKey) {
...
+  if (!urlBase64Helper.validate(publicKey)) {
     throw new Error('Vapid public key must be a URL safe Base 64 (without "=")');
   }
 
-  publicKey = urlBase64.decode(publicKey);
+  publicKey = urlBase64Helper.decode(publicKey);
 
   if (publicKey.length !== 65) {
-    throw new Error('Vapid public key should be 65 bytes long when decoded.');
+    throw new Error(`Vapid public key should be 65 bytes long when decoded. got (${publicKey.length}) ${publicKey}`);
```
Comment (marco-c) [inline src/vapid-helper.js:110]:
```suggestion
    throw new Error(`Vapid public key should be 65 bytes long when decoded, got (${publicKey.length}) ${publicKey}`);
```

File: .gitignore (hunk 1 above)
Comment (haouarihk) [reply .gitignore:9]: i see sorry

Comment (marco-c) [reply .gitignore:9]: No problem, thank you for working on the PR!

BTW, you don't need to close and create a new one, you can also just update an existing PR.

**Comment by hikarunoryoma** [comment]
Thanks for addressing here!

**Pull request closed without merging**